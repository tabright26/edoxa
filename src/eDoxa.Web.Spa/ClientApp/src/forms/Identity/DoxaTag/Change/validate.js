const validate = values => {
  const errors = {};
  if (!values.name) {
    errors.name = "DoxaTag required.";
  } else if (values.name.length < 2 && values.name.length > 16) {
    errors.name = "Must between 16 characters and greater than characters 2.";
  } else if (!/^[0-9a-zA-Z_]*$/.test(values.name)) {
    errors.name = "Invalid format.";
  }
  return errors;
};

export default validate;
