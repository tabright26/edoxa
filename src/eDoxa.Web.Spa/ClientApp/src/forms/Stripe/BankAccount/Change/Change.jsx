import React from "react";
import Button from "components/Shared/Override/Button";
import Input from "components/Shared/Override/Input";
import { Field, reduxForm } from "redux-form";
import { FormGroup, Form } from "reactstrap";
import { CREATE_BANK_ACCOUNT_FORM } from "forms";

const style = {
  backgroundColor: "#fff",
  color: "#000",
  borderColor: "#f1f1f1"
};

const ChangeBankAccountForm = ({ handleSubmit, handleCancel }) => (
  <Form onSubmit={handleSubmit}>
    <Field style={style} type="text" name="country" label="country" formGroup={FormGroup} component={Input.Text} />
    <Field style={style} type="text" name="accountHolderName" label="accountHolderName" formGroup={FormGroup} component={Input.Text} />
    <Field style={style} type="text" name="routingNumber" label="routingNumber" formGroup={FormGroup} component={Input.Text} />
    <Field style={style} type="text" name="accountNumber" label="accountNumber" formGroup={FormGroup} component={Input.Text} />
    <FormGroup className="mb-0">
      <Button.Save className="mr-2" />
      <Button.Cancel onClick={handleCancel} />
    </FormGroup>
  </Form>
);

export default reduxForm({ form: CREATE_BANK_ACCOUNT_FORM })(ChangeBankAccountForm);
