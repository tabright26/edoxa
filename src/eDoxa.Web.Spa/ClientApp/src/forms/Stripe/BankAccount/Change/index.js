import ChangeBankAccountForm from "./Change";

export default ChangeBankAccountForm;
