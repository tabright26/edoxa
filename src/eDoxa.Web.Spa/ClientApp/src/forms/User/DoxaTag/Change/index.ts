import UpdateDoxaTagForm from "./Change";

export default UpdateDoxaTagForm;
