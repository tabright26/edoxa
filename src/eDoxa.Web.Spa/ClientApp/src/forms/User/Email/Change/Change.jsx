import React from "react";
import { Field, reduxForm } from "redux-form";
import { FormGroup, Input, Form } from "reactstrap";
import Button from "components/Buttons";
import { CHANGE_EMAIL_FORM } from "forms";

const ChangeEmailForm = ({ handleSubmit, handleCancel }) => (
  <Form onSubmit={handleSubmit}>
    <FormGroup>
      <Field type="text" name="email" component={({ input }) => <Input {...input} placeholder="Email" bsSize="sm" />} />
    </FormGroup>
    <FormGroup className="mb-0">
      <Button.Save className="mr-2" />
      <Button.Cancel onClick={handleCancel} />
    </FormGroup>
  </Form>
);

export default reduxForm({ form: CHANGE_EMAIL_FORM })(ChangeEmailForm);
