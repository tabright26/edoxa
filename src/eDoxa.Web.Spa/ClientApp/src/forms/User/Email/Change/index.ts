import ChangeEmailForm from "./Change";

export default ChangeEmailForm;
