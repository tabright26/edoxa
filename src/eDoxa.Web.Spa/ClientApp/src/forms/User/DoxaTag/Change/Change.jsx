import React from "react";
import { reduxForm, Field } from "redux-form";
import { FormGroup, Form } from "reactstrap";
import Button from "../../../../components/Override/Button";
import Input from "../../../../components/Override/Input";
import validate from "./validate";
import { CHANGE_DOXATAG_FORM } from "../../../../forms";

const ChangeDoxaTagForm = ({ handleSubmit, handleCancel }) => (
  <Form onSubmit={handleSubmit}>
    <Field type="text" name="name" label="DoxaTag" formGroup={FormGroup} component={Input.Text} />
    <FormGroup className="mb-0">
      <Button.Save className="mr-2" />
      <Button.Cancel onClick={handleCancel} />
    </FormGroup>
  </Form>
);

export default reduxForm({ form: CHANGE_DOXATAG_FORM, validate })(ChangeDoxaTagForm);
