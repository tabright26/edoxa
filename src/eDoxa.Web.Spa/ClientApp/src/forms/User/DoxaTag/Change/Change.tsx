import React, { FunctionComponent } from "react";
import { FormGroup, Form } from "reactstrap";
import { reduxForm, Field } from "redux-form";
import Button from "components/Shared/Override/Button";
import Input from "components/Shared/Override/Input";
import { CHANGE_DOXATAG_FORM } from "forms";
import validate from "./validate";

const ChangeDoxaTagForm: FunctionComponent<any> = ({ handleSubmit, handleCancel }) => (
  <Form onSubmit={handleSubmit}>
    <Field type="text" name="name" label="DoxaTag" formGroup={FormGroup} component={Input.Text} />
    <FormGroup className="mb-0">
      <Button.Save className="mr-2" />
      <Button.Cancel onClick={handleCancel} />
    </FormGroup>
  </Form>
);

export default reduxForm<any, { handleCancel: () => any }, string>({ form: CHANGE_DOXATAG_FORM, validate })(ChangeDoxaTagForm);
