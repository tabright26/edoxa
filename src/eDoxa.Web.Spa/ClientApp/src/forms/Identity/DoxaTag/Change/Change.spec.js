import React from "react";
import Change from "./Change";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={{ getState: () => {}, dispatch: action => {}, subscribe: () => {} }}>
        <Change />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
