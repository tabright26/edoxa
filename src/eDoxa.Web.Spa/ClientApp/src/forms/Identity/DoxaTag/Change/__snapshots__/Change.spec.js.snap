// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <input
      className="form-control-sm form-control"
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="DoxaTag"
      type="text"
      value=""
    />
    <div
      className="invalid-feedback"
    />
  </div>
  <div
    className="mb-0 form-group"
  >
    <button
      aria-label={null}
      className="mr-2 btn btn-primary btn-sm"
      onClick={[Function]}
      type="submit"
    >
      Save
    </button>
    <button
      aria-label={null}
      className="btn btn-outline-primary btn-sm"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</form>
`;
